api/http/authn.go

```diff
@@ -40,6 +40,9 @@ func AuthenticateMiddleware(authn smqauthn.Authentication, domainCheck bool) fun
 				resp.DomainID = domain
```

Comment:
Can we have something like this ? 
```suggestion
				resp.DomainID = domain
				switch resp.Role {
				case smqauthn.AdminRole:
					resp.DomainUserID = resp.UserID
				case smqauthn.UserRole:
					resp.DomainUserID = auth.EncodeDomainUserID(domain, resp.UserID)
				} 
```